bitnami/grafana-operator/templates/grafana.yaml

```diff
@@ -96,6 +97,25 @@ spec:
           {{- if .Values.grafana.podSecurityContext.enabled }}
           securityContext: {{- omit .Values.grafana.podSecurityContext "enabled" | toYaml | nindent 12 }}
           {{- end }}
+          {{- if .Values.grafana.tolerations }}
+          tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.grafana.tolerations "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.grafana.nodeSelector }}
+          nodeSelector: {{ toYaml .Values.grafana.nodeSelector | nindent 12 }}
+          {{- end }}
+          {{- if .Values.grafana.extraVolumes | or .Values.grafana.secrets }}
+          volumes:
+            {{- range .Values.grafana.secrets }}
+            - name: '{{.}}'
+              secret:
+                defaultMode: 420
+                optional: true
+                secretName: '{{.}}'
+            {{- end }}
```

Comment:
What about using a similar approach as for example elasticsearch? There we expect this logic to be defined in the provided values.

```
## extraVolumes:
##   - name: es-certs
##     secret:
##       defaultMode: 420
##       secretName: es-certs
```